frinx-uniconfig/user-guide/operational-procedures/uniconfig-properties/readme.md

```diff
@@ -0,0 +1,89 @@
+# Uniconfig Properties
+
+## Introduction
+
+UniConfig has input parameters to configure it precisely.
+All these parameters are located in the *application.properties* file.
+These parameters are divided into 2 groups - immutable parameters, 
+which are the majority, and mutable parameters, which can be
+changed in runtime without restarting UniConfig with *UniConfig Cloud Config*.
+
+Mutable properties consist of *crypto*, *schema-settings*, *callbacks*,
+*notifications.kafka*, *netconf-default-parameters*, *gnmi-default-parameters*
+and *cli-default-parameters* property prefixes. These properties are stored
+in the database in the table *properties* and are known as *default properties*.
+User can read *default properties* with *read-properties RPC* and update it with *update-properties RPC*.
+
+!!!
+After starting UniConfig, if the database contains these properties, 
+UniConfig will use the values of these properties from the database. But if these 
+parameters are not in the database, then the values from first UniConfig instance 
+right after its startup will be used (by application.properties file or env variables) 
+and saved to the database for the next UniConfig instances.
+!!!
+
+## UniConfig Cloud Config
+
+UniConfig Cloud Config is pretty the same technology as Spring Cloud Config with JDBC backend:
+https://docs.spring.io/spring-cloud-config/reference/4.1/server/environment-repository/jdbc-backend.html 
+and Spring Cloud Bus: https://docs.spring.io/spring-cloud-bus/docs/current/reference/html/.
+
+The main difference between these two, is that UniConfig Cloud Config Server and 
+Cloud Config Client are in the same project, while Spring requires the use of a separate
+Cloud Config Server application.
+
+It works as follows:
+
+![startup-with-ucc](startup-with-ucc.jpg)
```

Comment:
In the diagram, UC reads default props -> calls refresh -> reads default propps -> calls refresh

^ this seems like a loop, could you explain why the second refresh does not trigger the full refresh again ?

Reply:
because of few UC startup in the same time (crypto properties) and its done only while start-up, in runtime we have only 1 refresh by update RPC